qmxgraph/page/api.js

```diff
@@ -453,6 +471,54 @@ graphs.Api.prototype.getGeometry = function getGeometry (cellId) {
 };
 
 /**
+ * Gets the decoration's relative position.
+ *
+ * @param {number} cellId Id of a decoration in graph.
+ * @returns {number} Returns an a normalized number between [0, 1] representing the position of the
+ * decoration along the parent edge.
+ * @throws {Error} Unable to find the cell or it is not an decoration.
+ */
+graphs.Api.prototype.getDecorationPosition = function getDecorationPosition (cellId) {
+    "use strict";
+
+    var graph = this._graphEditor.graph;
```

Comment:
Please move this code block:

```js
     var graph = this._graphEditor.graph;
     var model = graph.getModel();
     var cell = this._findCell(model, cellId);
     if (!cell.isDecoration()) {
         throw Error("The cell " + cellId + " is not a decoration");
     }
```

To its own function and reuse it; this code seems to be copied over a number of functions